exchanges/graphcache/src/cacheExchange.ts

```diff
@@ -213,11 +213,8 @@ export const cacheExchange = <C extends Partial<CacheExchangeOpts>>(
       optimisticKeysToDependencies.delete(operation.key);
     }
 
-    reserveLayer(
-      store.data,
-      operation.key,
-      operation.kind === 'subscription' || result.hasNext
-    );
+    if (operation.kind === 'subscription' || result.hasNext)
+      reserveLayer(store.data, operation.key, true);
```

Comment:
This meant that we were:
- unnecessarily adding layers to `optimisticOrder` even when that wasn't needed
- where shifting non-deferred layers that already were registered before, which is unexpected. Our tests are relying on this though, so we can "cheat" here